api/src/dataloaders/mint_history.rs

```diff
@@ -0,0 +1,60 @@
+use std::collections::HashMap;
+
+use async_graphql::{dataloader::Loader as DataLoader, FieldError, Result};
+use poem::async_trait;
+use sea_orm::{prelude::*, JoinType, Order, QueryOrder, QuerySelect};
+
+use crate::{
+    db::Connection,
+    entities::{collection_mints, mint_history},
+};
+
+#[derive(Debug, Clone)]
+pub struct CollectionLoader {
+    pub db: Connection,
+}
+
+impl CollectionLoader {
+    #[must_use]
+    pub fn new(db: Connection) -> Self {
+        Self { db }
+    }
+}
```

Comment:
CollectionMintHistoryLoader

I struggle with the loader names does it make sense to name them after what they do?